docs/GuidGuidRecommender.md

```diff
@@ -0,0 +1,144 @@
+
+# Overview
+
+The GUID-GUID coinstallation recommender system generates recommendations
+for add-ons which are similar to a given add-on,
+where similarity is based on the number of profiles
+which have both add-ons installed.
+
+As recommendations are only surfaced
+for a specific set of add-ons meeting certain criteria,
+and depend only on characteristics of the add-ons themselves,
+recommendation sets for each add-on can be (and are) computed in advance.
+
+Recommendations are generated as follows:
+
+1. The [input data](#input-data),
+    consisting of coinstallation counts for each pair of add-ons,
+    is loaded from a precomputed file.
+2. For each add-on, we determine a [__candidate list__](#generating-the-scored-candidate-lists) of "related" add-ons,
+    together with associated __relevance scores__,
+    based on the coinstallation data.
+3. Recommendations for a given add-on are [selected](#selecting-recommendations)
+    by retaining the N add-ons in the candidate list
+    with the highest relevance scores.
+
+Once the recommender has been trained,
+we apply various [metrics](./GraphRepresentation.md#quality-and-health-metrics)
+to the coinstallation data, candidate lists, and recommendations,
+to evaluate the overall health of the system
+and the quality of recommendations.
+
+The recommender system is implemented by the [`GuidGuidCoinstallRecommender`](../taar_lite/recommenders/guidguid.py#L10) class,
+which is instantiated by the [TaarLiteAppResource](../taar_lite/app/production.py#L44).
+
+Further technical details are provided on [this page](./GraphRepresentation.md).
+
+
+# Input data
+
+The primary data source used in generating recommendations
+is the file `guid_coinstallation.json`,
```

Comment:
hmmm....`GuidGuidCoinstallRecommender` doesn't care what you call the file.

I'd rather separate out the general from the deployment specifics.

So, I'm interested in the format of the input data that GuidGuidCoinstallRecommender expects and what it means.

That, in production it comes from an ETL job, and is lazily loaded through a cache every 4 hours, etc, etc belongs in full documentation about the deployment / production instance.